book/08-customizing-git/sections/hooks.asc

```diff
@@ -111,6 +111,14 @@ The first script to run when handling a push from a client is `pre-receive`.
 It takes a list of references that are being pushed from stdin; if it exits non-zero, none of them are accepted.
 You can use this hook to do things like make sure none of the updated references are non-fast-forwards, or to do access control for all the refs and files they're modifying with the push.
 
+===== `proc-receive`
+
+The `proc-receive` script is quite different from other server-side scripts, it is invoked by git push process, but can change the worklow or process between `pre-reveive` and
+`post-receive` script. This's because if the upcoming update references match the specified reference pattern, which can be configured by `receive.procReceiveRefs`, then `proc-receive`
+will take over the whole flow after `pre-receive`. It can decide to create, update or rename new reference, or even fallback to previous process.
+In other word, it can outsource reference processing flow, this give a system administrator to customize very useful policy for pushing workflow, such as Gerrit-like workflow in Git.
```

Comment:
```suggestion
In other words, it can outsource reference processing flow, this gives a system administrator to customize a very useful policy for pushing workflow, such as Gerrit-like workflow in Git.
```

Additionally, please, follow the rule: 1 sentence per 1 line.